src/lib/utils/meta_tables/abstract_meta_system_table.cpp

```diff
@@ -0,0 +1,55 @@
+#ifdef __linux__
+
+#include <fstream>
+
+#endif
+
+#ifdef __APPLE__
+
+#include <mach/mach.h>
+
+#endif
+
+#include "utils/meta_tables/abstract_meta_system_table.hpp"
+
+namespace opossum {
+
+AbstractMetaSystemTable::AbstractMetaSystemTable(const TableColumnDefinitions& column_definitions)
+    : AbstractMetaTable(column_definitions) {}
+
+int AbstractMetaSystemTable::_get_cpu_count() {
+#ifdef __linux__
+
+  std::ifstream cpu_info_file;
+  cpu_info_file.open("/proc/cpuinfo", std::ifstream::in);
+
+  if (!cpu_info_file.is_open()) {
+    Fail("Unable to open /proc/cpuinfo");
+  }
+
+  uint32_t processors = 0;
+  std::string cpu_info_line;
+  while (std::getline(cpu_info_file, cpu_info_line)) {
+    if (cpu_info_line.rfind("processor", 0) == 0) ++processors;
+  }
+
+  cpu_info_file.close();
+
+  return processors;
+#endif

```

Comment:
Which also works on Mac. Not sure if by default or because of some package though.

Reply:
it's part of coreutils. 
however some other key figures could also be obtained by calling an external command but i thought it would be cleaner to not rely on external tools. what's the general opinion here? avload, numactl or free could make a lot of proc file parsing obsolete.

Reply:
Having a widely available solution might not be too bad.